### ISSUE TITLE
Face bone positons/rotations corrupt when importing animations

### ISSUE THREAD

**Issue by Spantle** [DarklightGames/io_scene_psk_psa#85]
**Versions**
Blender: 4.0.2
io_scene_psk_psa: 6.2.0

**Describe the bug**
Upon exporting the character animations from UModel and importing them into Blender, the face bones appear to be offset and corrupt. Resetting the face bones to their default locations/rotations fixes them and restores the default face, but this means that the facial animations can't be restored.

I've tried subtracting the offsets, moving bones manually, and a whole heap of other things, but the facial animations seem unrecoverable. Is there anything that can be done?

**To Reproduce**
1. Export the model and animations from UModel
2. Import the model animations into Blender
3. Apply the Action to the model

**Screenshots**
Here's a normal animation in UModel (animation 1):
![umodel_acl_1 3_2024-03-23_19-24-29](https://github.com/DarklightGames/io_scene_psk_psa/assets/79182061/829ea63e-1093-42ee-acb2-365a0ac35ee3)
Here's a facial animation in UModel (animation 2):
![umodel_acl_1 3_2024-03-23_19-34-43](https://github.com/DarklightGames/io_scene_psk_psa/assets/79182061/3b94235b-6ea2-4e38-92ed-7b9cc904ba92)
Here's animation 1 in Blender:
![blender_2024-03-23_19-28-28](https://github.com/DarklightGames/io_scene_psk_psa/assets/79182061/4467df70-28fd-4bb5-880f-3138214b1fd5)
Here's animation 2 in Blender:
![blender_2024-03-23_19-34-09](https://github.com/DarklightGames/io_scene_psk_psa/assets/79182061/a1701a09-7b49-40b8-bf89-fc3692c45b80)

**Files**
Please post attachments of the PSK and/or PSA files that you are using to encounter the bug, if any.
I would've uploaded the files directly, but GitHub does not allow PSK or PSA files to be uploaded (for some reason?) so I have zipped them. I hope this is OK.
[PSK and PSAs.zip](https://github.com/DarklightGames/io_scene_psk_psa/files/14730880/PSK.and.PSAs.zip)

**Additional context**
These animations were exported using a special version of UModel [found here](https://www.gildor.org/smf/index.php?topic=8304.0) as the game uses an older version of ACL (1.3).

**Comment by cmbasnett** [comment]
Issue #83 reports the same thing about face bones going wild. Is there a warning about missing bones when the PSA is imported? I'm always a little wary when people start using experimental builds of UModel, there's no telling if the export is just simply incorrect or if there's some special flags I'm not reading or something.

**Comment by Spantle** [comment]
> Issue #83 reports the same thing about face bones going wild. Is there a warning about missing bones when the PSA is imported? I'm always a little wary when people start using experimental builds of UModel, there's no telling if the export is just simply incorrect or if there's some special flags I'm not reading or something.

I thought the same thing hence why [I thought to raise the issue to the UModel team](https://www.gildor.org/smf/index.php/topic,8729.new.html#new) and they helped me confirm that the issue is in fact with this importer plugin. By using [this older unmaintained importer plugin](https://github.com/DarklightGames/io_scene_psk_psa) the results are definitely much much better (although still not perfect possibly due to retargeting or an issue with the plugin, I'm not certain)

**Comment by Spantle** [comment]
I found and tried [another plugin](https://github.com/matyalatte/blender3d_import_psk_psa), and this one appears to produce seemingly perfect results. I hope you can use it as a reference to maybe fix your plugin since yours has better features overall.

**Comment by cmbasnett** [comment]
I figured out the issue here, it's actually a relatively easy fix.

The issue is that the PSK and the PSA bones are named the same, but the casing differs (i.e., `root` and `Root`, `face` and `Face` etc.) There's a way to tell it to match the bone names ignoring case, but there's a bug with it because I hadn't tested it properly after making some changes. I really need to add some unit tests or something!

In addition, I've fixed the body itself being flipped all over the place. I will push out a fix for this as soon as I can.

![image](https://github.com/DarklightGames/io_scene_psk_psa/assets/5035660/d4c014d4-3e86-4606-93ec-510358bb2967)
